packages/adblocker/src/filters/cosmetic.ts

```diff
@@ -630,6 +701,22 @@ export default class CosmeticFilter implements IFilter {
       }
     }
 
+    if (this.parentDomains !== undefined) {
+      const { hostnames, entities } = this.parentDomains;
+
+      if (hostnames !== undefined) {
+        for (const hostname of hostnames) {
+          tokens.push(new Uint32Array([hostname]));
+        }
+      }
+
+      if (entities !== undefined) {
+        for (const entity of entities) {
+          tokens.push(new Uint32Array([entity]));
+        }
+      }
+    }
```

Comment:
When the engine matches the filter in the bucket, it retrieves tokens from the filter and quickly selects the corresponding possible targets in the bucket. For an instance, a filter will be picked up if the filter has tokens of `A` and `B` and engine looked up for `A`. Here, we register `parentDomains` values so the filter can be fall into possible match candidates with those values.